README.md

```diff
@@ -48,161 +146,276 @@ To run the demo code, follow these steps:
+It requires the table to have a column named "id".
 
 ```js
 try {
-  const connection = await mysql.getConnection();
-  // Use the connection.
-  connection.release();
+  const rows = await connection.getById('flavors', 7);
+  // Process data in the result.
 } catch (e) {
   // Handle the error.
 }
 ```
 
-## `insert`
-This inserts a record into a given table
-and returns the id of the new record.
-The keys of obj are column names
-and their values are the values to insert.
-This assumes that the table has a column
-named `id` that is auto_increment.
+#### `insert`
+This inserts a record into a given table and returns the id of the new record.
+The keys of obj are column names and their values are the values to insert.
+This assumes that the table has a column named `id` that is auto_increment.
 
 ```js
 try {
-  const newId = await mysql.insert('flavors', {name: 'vanilla', calories: 100});
+  const newId = await connection.insert('flavors', {name: 'vanilla', calories: 100});
   // Do something after successful insert.
   // newId will be the id of the newly-inserted row.
 } catch (e) {
   // Handle the error.
 }
 ```
 
-## `query`
+#### `query`
 This executes a SQL query.
 It is the most general purpose function provided.
 It is used by several of the other functions.
 
 ```js
 try {
-  const rows = await mysql.query('select name from flavors where calories < 150');
+  const rows = await connection.query('select name from flavors where calories < 150');
   // Process data in the array rows.
 } catch (e) {
   // Handle the error.
 }
 
 try {
-  const sql = 'select name from flavors where calories < $1 and cost < $2';
-  const rows = await mysql.query(sql, 200, 3);
+  const sql = 'select name from flavors where calories < ? and cost < ?';
+  const rows = await connection.query(sql, 200, 3);
   // Process data in the array rows.
 } catch (e) {
   // Handle the error.
 }
 ```
 
-## `transaction`
+#### `transaction`
 This executes a given function inside a transaction.
-This function can call other functions in this API
-to perform database operations.
-The function must return a promise to indicate
-when it has completed (`resolve`)
-or when an error has occurred (`reject`).
-If the function throws an error or rejects,
-the transaction is rolled back.
-Otherwise it is committed.
+This function can call other functions in this API to perform database operations.
+The function must return a promise to indicate when it has completed (`resolve`) or when an error has occurred (`reject`).
+If the function throws an error or rejects, the transaction is rolled back.
+Otherwise, it is committed.
+
+```js
+try {
+  await connection.transaction(someFunction);
+  // Do more work after the transaction commits.
+} catch (e) {
+  // Handle the error.
+}
+```
 
 ```js
 try {
-  await mysql.transaction(someFunction);
+  await connection.transaction(() => {
+    return connection.insert('flavors', {name: 'vanilla', calories: 100}).then(id => {
```

Comment:
Is there a reason to prefer using `then` here instead of making the arrow function passed to transaction be async and using awaits?

Reply:
Not really.  I was more doing it because I felt like I wasn't doing enough examples using the traditional Promise API.  Given your comment above, I'm not worried about that any long :)